src/legacy/core_plugins/kibana/public/dashboard/migrations/migrations_730.ts

```diff
@@ -38,8 +39,36 @@ export function migrations730(
     doc.attributes.kibanaSavedObjectMeta.searchSourceJSON = JSON.stringify(
       moveFiltersToQuery(searchSource)
     );
+  } catch (e) {
+    // eslint-disable-next-line
+    console.error(
+      'Migration error encountered while trying to migrate queries out of filters array: ',
+      e
+    );
     return doc;
+  }
+
+  let uiState = {};
+  // Ignore errors, at some point uiStateJSON stopped being used, so it may not exist.
+  if (doc.attributes.uiStateJSON && doc.attributes.uiStateJSON !== '') {
+    uiState = JSON.parse(doc.attributes.uiStateJSON);
+  }
+
+  try {
+    const panels = JSON.parse(doc.attributes.panelsJSON);
+    doc.attributes.panelsJSON = JSON.stringify(
+      migratePanelsTo730(panels, '7.3.0', doc.attributes.useMargins, uiState)
+    );
+
+    delete doc.attributes.uiStateJSON;
   } catch (e) {
+    // eslint-disable-next-line
+    console.error(
+      'Migration error encountered while trying to migrate dashboard panels: ',
+      e
+    );
     return doc;
```

Comment:
Same here. Kibana will think the docs are all up to date, but you'll be dealing with docs in some invalid state.

Reply:
yea, look in migrations.js, everywhere in there that strange data is encountered, it just returns the doc. `// Let it go, the data is invalid and we'll leave it as is`.

I'm totally for changing this, but was wary of changing it in this same PR since I think it might need a little more thought and focus on it's own right.